libvast/vast/address_synopsis.hpp

```diff
@@ -27,6 +27,21 @@
 
 namespace vast {
 
+namespace {
+
+// Because VAST deserializes a synopsis with empty options and
+// construction of an address synopsis fails without any sizing
+// information, we augment the type with the synopsis options.
+type annotate_type(type type, const bloom_filter_parameters& params) {
+  using namespace std::string_literals;
+  auto v = "bloomfilter("s + std::to_string(*params.n) + ','
+           + std::to_string(*params.p) + ')';
+  // Replaces any previously existing attributes.
+  return std::move(type).attributes({{"synopsis", std::move(v)}});
+}
```

Comment:
Not in this PR, but need a better solution for this type tagging.

Reply:
Indeed, I think we say this in every PR where we touch this code ;)